src/main/java/HttpUrlConnection.java

```diff
@@ -4,14 +4,14 @@
 import java.net.http.HttpResponse;
 
 public class HttpUrlConnection {
-    public void getAllcurrency() throws Exception {
+    public String getAllcurrency() throws Exception {
```

Comment:
Instead of string return defined response model

Reply:
So here you returning `String` instead CurrencyResponse model that I have suggested to have instead

Reply:
Create CurrencyResponse.java which is same as response you get, map and then return that currency response object